### PR TITLE
Add Bitcoin bech32 address

### PR DESCRIPTION
**Pull request by hewigovens**
support public key to bech32 p2wpkh address.

### CODE REVIEW

File: Sources/Slip.swift [hunk 1]
```diff
@@ -23,3 +23,29 @@ public enum Slip: Int {
...
+        switch self {
+        case .bitcoin:
+            return "bc"
+        case .litecoin:
+            return "ltc"
+        default:
+            return ""
+        }
+    }
+    var hrpTestnet: String {
```
Comment (vikmeup) [inline Sources/Slip.swift]: let's remove test net stuff, no need

File: Sources/Bitcoin/BitcoinSegwitAddress.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                ret.append(UInt8((acc << (to - bits)) & maxv))
+            }
+        } else if bits >= from || ((acc << (to - bits)) & maxv) != 0 {
+            return nil
+        }
+        return ret
+    }
+}
+
+public struct BitcoinSegwitAddress: Address {
```
Comment (vikmeup) [inline Sources/Bitcoin/BitcoinSegwitAddress.swift]: can you move BitcoinSegwitAddress into separate file

File: Sources/Slip.swift [hunk 3]
```diff
@@ -23,3 +23,29 @@ public enum Slip: Int {
     case callisto = 820
     case wanchain = 5718350
 }
+
+///  Registered human-readable parts for BIP-0173
+///
+/// - SeeAlso: https://github.com/satoshilabs/slips/blob/master/slip-0173.md
+public extension Slip {
+    var hrp: String {
```
Comment (vikmeup) [inline Sources/Slip.swift]: can we make switch exhaustive?

File: Sources/Crypto.m [hunk 4]
```diff
@@ -120,6 +120,30 @@ + (NSData *)base58Decode:(nonnull NSString *)string {
...
+// MARK: - Bech32
+
++ (nonnull NSString *)bech32Encode:(nonnull NSData *)data hrp:(nonnull NSString *)hrp
+{
+    NSMutableData *result = [[NSMutableData alloc] initWithCapacity:89];
+    bech32_encode(result.mutableBytes, hrp.UTF8String, data.bytes, data.length);
+    return [NSString stringWithUTF8String:result.bytes];
+}
+
++ (nullable NSData *)bech32Decode:(nonnull NSString *)string hrp:(NSString * _Nullable *)hrp
```
Comment (vikmeup) [inline Sources/Crypto.m]: can we add a test for these two functions?

File: Sources/Bitcoin/WitnessProgram.swift [hunk 5]
```diff
@@ -12,6 +12,7 @@ public struct WitnessProgram: Equatable {
     public let program: Data
 
     var encoded: Data {
```
Comment (vikmeup) [inline Sources/Bitcoin/WitnessProgram.swift]: Should this be optional? If no valid

Comment (hewigovens) [reply Sources/Bitcoin/WitnessProgram.swift]: here nil is same semantic as empty Data()

**Pull request merged**